### PR TITLE
cli: use json logs on non-terminal envs (close #2770)

### LINKED ISSUE #2770: Option for Simple Logs
Hello!

We're using a customized version of the`hasura/graphql-engine:v1.0.0-beta.3.cli-migrations` docker container to automate hasura migrations in AWS.

Can we add support for an option for simpler logging? The current logs are hard to read in AWS
```
Checking for update... ◐  Checking for update... ◓  Checking for update... ◑  Checking for update... ◒  Checking for update... ◐  Checking for update... ◓  Checking for update... ◑  Checking for update... ◒  Checking for update... ◐  Checking for update... ◓  Checking for update... ◑  Applying migrations...◐  Applying migrations...◓  
```

### PR DESCRIPTION
**Pull request by ashishra0**
Issue #2770 
Added JSON formatter to display simple logs for non-cli environments. Add conditional rendering for spinner based on CLI environments.

### CODE REVIEW

File: cli/cli.go [hunk 1]
```diff
@@ -182,6 +182,8 @@ type ExecutionContext struct {
 
 	// SkipUpdateCheck will skip the auto update check if set to true
 	SkipUpdateCheck bool
+
+	isEnv bool
```
Comment (shahidhk) [inline cli/cli.go]:
```suggestion
	isRunningOnDocker bool
```

File: cli/cli.go [hunk 2]
```diff
@@ -212,6 +214,12 @@ func (ec *ExecutionContext) Prepare() error {
 	// populate version
 	ec.setVersion()
 
+	// check environment
+	val, _ := os.LookupEnv("HASURA_GRAPHQL_CLI_ENVIRONMENT")
```
Comment (shahidhk) [inline cli/cli.go]: Move this to `readConfig` and use viper instead of `os.LookupEnv`

File: cli/cli.go [hunk 3]
```diff
@@ -322,6 +324,12 @@ func (ec *ExecutionContext) readConfig() error {
 		Endpoint:    v.GetString("endpoint"),
 		AdminSecret: adminSecret,
 	}
+
+	// check environment
+	val := v.GetString("HASURA_GRAPHQL_CLI_ENVIRONMENT")
```
Comment (shahidhk) [inline cli/cli.go]:
```suggestion
	val := v.GetString("cli_environment")
```

**Comment by shahidhk** [comment]
@ashishra0 Let's use this method to detect if we have a terminal or not:

```go
package main
 
import (
    "os"
    "fmt"
    "golang.org/x/crypto/ssh/terminal"
)
 
func main() {
    if terminal.IsTerminal(int(os.Stdout.Fd())) {
        fmt.Println("Hello terminal")
    } else {
        fmt.Println("Who are you?  You're not a terminal.")
    }
}
```

If we have a terminal (`terminal.IsTerminal`), use textformatter, otherwise use JSON formatter and disable spinner.

File: cli/cli.go [hunk 4]
```diff
@@ -337,25 +338,34 @@ func (ec *ExecutionContext) setupSpinner() {
...
 // Spin stops any existing spinner and starts a new one with the given message.
 func (ec *ExecutionContext) Spin(message string) {
-	ec.Spinner.Stop()
-	ec.Spinner.Prefix = message
-	ec.Spinner.Start()
+	if terminal.IsTerminal(int(os.Stdout.Fd())) {
+		ec.Spinner.Stop()
+		ec.Spinner.Prefix = message
+		ec.Spinner.Start()
+	}
```
Comment (shahidhk) [inline cli/cli.go]: Missed an else condition here.

**Comment by mwarger** [comment]
@shahidhk I was curious if there is a way to use this before it's released?  Or how I can enable this ability to better see logs (I'm using this for stackdriver).  thank you!

**Comment by shahidhk** [comment]
@mwarger We'll be making 1.1 release hopefully today. If you'd still want to try it out earlier, you can build a version using `VERSION=dev make build` in the `cli` directory.

**Pull request merged**